### PR TITLE
Allowed to add remote assets

### PR DESCRIPTION
**Pull request by aleeks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Allowed to add remote assets
| Type?         | improvement
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | http://forge.prestashop.com/browse/BOOM-1837
| How to test?  | See : 

Add this in one controller : 

`$this->registerJavascript('remote-bootstrap-head', 'https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js', ['server' => 'remote', 'position' => 'head', 'priority' => 20]);`
`$this->registerJavascript('remote-bootstrap', 'https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js', ['server' => 'remote', 'position' => 'bottom', 'priority' => 20]);`
`$this->registerStylesheet('remote-bootstrap', 'https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css', ['server' => 'remote', 'priority' => 20]);`

And see your front

See last comment :https://github.com/PrestaShop/PrestaShop/pull/6656
ping @jorgevrgs

### CODE REVIEW

**Comment by mbrentini** [comment]
Hi,

IMHO this PR is critical. Many modules rely on external assets and these are broken for now.

Thank you in advance for reviewing this

File: classes/assets/StylesheetManager.php [hunk 1]
```diff
@@ -42,15 +42,23 @@ class StylesheetManagerCore extends AbstractAssetManager
...
 
-    public function register($id, $relativePath, $media = self::DEFAULT_MEDIA, $priority = self::DEFAULT_PRIORITY, $inline = false)
-    {
-        if ($fullPath = $this->getFullPath($relativePath)) {
-            $this->add($id, $fullPath, $media, $priority, $inline);
+    public function register(
+        $id,
+        $relativePath,
+        $media = self::DEFAULT_MEDIA,
+        $priority = self::DEFAULT_PRIORITY,
```
Comment (mickaelandrieu) [inline classes/assets/StylesheetManager.php]: Theses class seems identical. Can we use an abstract class and rely on it for the `register` and `add` functions ?

File: classes/controller/FrontController.php [hunk 2]
```diff
@@ -1032,11 +1033,12 @@ public function registerJavascript($id, $relativePath, $params = array())
             'priority' => AbstractAssetManager::DEFAULT_PRIORITY,
             'inline' => false,
             'attributes' => null,
+            'server' => 'local',
```
Comment (mickaelandrieu) [inline classes/controller/FrontController.php]: same here

**Comment by mickaelandrieu** [comment]
Thank you @aleeks

**Pull request merged**